parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -35,31 +38,79 @@ cleanupContainerdDlFiles() {
     rm -rf $CONTAINERD_DOWNLOADS_DIR
 }
 
-installContainerRuntime() {
-    if [ "${NEEDS_CONTAINERD}" == "true" ]; then
-        echo "in installContainerRuntime - KUBERNETES_VERSION = ${KUBERNETES_VERSION}"
-        local containerd_version
-        if [ -f "$MANIFEST_FILEPATH" ]; then
-            containerd_version="$(jq -r .containerd.edge "$MANIFEST_FILEPATH")"
-            if [ "${UBUNTU_RELEASE}" == "18.04" ]; then
-                containerd_version="$(jq -r '.containerd.pinned."1804"' "$MANIFEST_FILEPATH")"
-            fi
-        else
-            echo "WARNING: containerd version not found in manifest, defaulting to hardcoded."
-        fi
+# After the centralized packages changes, the containerd versions are only available in the components.json.
+installContainerdWithComponentsJson() {
+    os=${UBUNTU_OS_NAME}
+    if [[ -z "$UBUNTU_RELEASE" ]]; then
+        os=${MARINER_OS_NAME}
+        os_version="current"
+    fi
+    os_version="${UBUNTU_RELEASE}"
+    containerdPackage=$(jq ".Packages" "$COMPONENTS_FILEPATH" | jq ".[] | select(.name == \"containerd\")") || exit $ERR_CONTAINERD_VERSION_INVALID
+    PackageVersions=()
+    returnPackageVersions "${containerdPackage}" "${os}" "${os_version}"
```

Comment:
somewhat confused... how is `returnPackageVersions` populating `PackgeVersions`?

Reply:
I did't find a good way to return an array of strings in Bash scripts. So the func `returnPackageVersions` is actually modifying the global variable `PackageVersions` , defined 1 line above,